docs/reference/docs/reindex.asciidoc

```diff
@@ -0,0 +1,461 @@
+[[docs-reindex]]
+==== Reindex API
+
+`_reindex`'s most basic form just copies documents from one index to another.
```

Comment:
I guess it's fine to have them here we can still catch up with #16618 once it's in?